packages/gelato-adapter/src/adapter.ts

```diff
@@ -51,11 +61,40 @@ export const adaptGelatoRelayer = (
       }
 
       try {
-        const relayResponse = await relay.sponsoredCallERC2771(
-          request,
-          signer.provider as any,
-          gelatoApiKey
-        )
+        let relayResponse: RelayResponse
+
+        if (gelatoApiKey) {
+          relayResponse = await relay.sponsoredCallERC2771(
+            request,
+            signer.provider as any,
+            gelatoApiKey
+          )
+        } else if (gelatoProxyApiUrl) {
+          const signatureData = await relay.getSignatureDataERC2771(
+            request,
+            signer.provider as any,
+            ERC2771Type.SponsoredCall
+          )
+          // userNonce must be a number to be parsed into JSON
+          const signatureDataFormatted = {
+            ...signatureData,
+            struct: {
+              ...signatureData.struct,
+              userNonce: Number(signatureData.struct.userNonce),
+            },
+          }
+          const response = await fetch(gelatoProxyApiUrl, {
+            method: 'POST',
+            headers: {
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({ signatureData: signatureDataFormatted }),
+          })
+          relayResponse = await response.json()
+        } else
```

Comment:
Can we wrap the else in `{}` just to make it more readable?

Reply:
Yes, that will be the linter that has made it unreadable 😆